subsystem/user_member_guild/fetch_member_guild_ranking_one_term.go

```diff
@@ -2,17 +2,30 @@ package user_member_guild
 
 import (
 	"elichika/client"
+	"elichika/subsystem/cache"
 	"elichika/userdata"
 	"elichika/utils"
 
 	"sort"
 )
 
+var (
+	getFetchMemberGuildRankingOneTermCache = cache.UniquePointerMap[int64, cache.CachedObject[client.MemberGuildRankingOneTerm]]{}
+)
+
 func FetchMemberGuildRankingOneTerm(session *userdata.Session, memberGuildId int32) client.MemberGuildRankingOneTerm {
-	// TODO(cache)
-	// official server only return 10 of the top ranking unless it's the current one
-	// it's likely that this was done using a table for total point and a SELECT ORDER BY LIMIT sql
-	// we build the full result for now, the truncation can be done by the caller
+	key := (int64(memberGuildId) << 32)
+	cacher := getFetchMemberGuildRankingOneTermCache.Get(key)
+	cacher.Acquire()
+	defer cacher.Release()
+	if cacher.ExpireAt <= session.Time.Unix() {
```

Comment:
If `memberGuildId` is not the current one, then the result for this shouldn't change.

So if we take things to the extreme, the most optimal cache would be something like:
- Save the last time the result was calculated.
- Calculate again if this time is < the time when that member guild period end.
- Maybe there's a limit on the time too.

To do this with the generic cacher, maybe just check if `memberGuildId` is the last one and set a max expire time?